src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -100,7 +101,15 @@ public async Task RunAsync()
             StepHost.OutputDataReceived += OnDataReceived;
             StepHost.ErrorDataReceived += OnDataReceived;
 
-            string file = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Externals), "node", "bin", $"node{IOUtil.ExeExtension}");
+            bool useNode10 = ExecutionContext.Variables.GetBoolean("AGENT_USE_NODE10_FOR_NODE_TASKS") ??
+                StringUtil.ConvertToBoolean(System.Environment.GetEnvironmentVariable("AGENT_USE_NODE10_FOR_NODE_TASKS"), false);
+            
+            string nodeFolder = (Data is Node10HandlerData || useNode10) ? "node10" : "node";
+
+            Trace.Info($"Task provided node10 handler data: {Data is Node10HandlerData}, Use node10 for node tasks: {useNode10}");
```

Comment:
we don't have to trace here, the process invoker will trace out the file name for you by default.

Reply:
I added the trace because I figured that it'd be useful during investigations to know the reasons why the handler chose to use the node10 backend. Otherwise, you'd need to either inspect the task.json manually.